examples/multiphase_flow/ex12/LSLocateStructureInterface.cpp

```diff
@@ -0,0 +1,271 @@
+        {
+            Pointer<Patch<NDIM> > patch = level->getPatch(p());
+            Pointer<CellData<NDIM, double> > D_data = patch->getPatchData(D_idx);
+            D_data->fill(0.0);
+        }
+
+        // Set lag data to -1.0
+        if (!d_lag_data_manager->levelContainsLagrangianData(ln)) continue;
+
+        Vec petsc_vec = lag_phi[ln]->getVec();
+        VecSet(petsc_vec, 1.0 * d_vol_elem);
+    }
+
+    std::vector<Pointer<LData> > PHI_data(finest_ln + 1, Pointer<LData>(NULL));
+    PHI_data[finest_ln] = lag_phi[finest_ln];
+    d_lag_data_manager->spread(D_idx, PHI_data, X_data, (RobinPhysBdryPatchStrategy*)NULL);
+    return;
+} // setLevelSetPatchDataBySpreading
+
+void
+LSLocateStructureInterface::setLevelSetPatchDataByGeometry(int D_idx,
+                                                           Pointer<HierarchyMathOps> hier_math_ops,
+                                                           double time,
+                                                           bool initial_time)
+{
+    // In this version of this class, the initial level set location is set to be
+    // exact since we always know the radius of the ball
+
+    static const double m = tan(d_wedge->wedge_angle);
+    static const double wedge_height = 0.5 * (d_wedge->wedge_length) * m;
+
+#if (NDIM == 3)
+    // Normal of right plane
+    static const double ar = -sin(d_wedge->wedge_angle);
+    static const double br = 0.0;
+    static const double cr = cos(d_wedge->wedge_angle);
+
+    // Normal of left plane
+    static const double al = sin(d_wedge->wedge_angle);
+    static const double bl = 0.0;
+    static const double cl = cos(d_wedge->wedge_angle);
+#endif
+
+    Pointer<PatchHierarchy<NDIM> > patch_hierarchy = hier_math_ops->getPatchHierarchy();
+    const int coarsest_ln = 0;
+    const int finest_ln = patch_hierarchy->getFinestLevelNumber();
+
+    // Set the initial condition for locating the interface
+    IBTK::Vector& X0 = d_wedge->X0;
+
+    // Get the coordinate of the bottom point of the wedge
+    X0(NDIM - 1) = getMinimumWedgeCoord(hier_math_ops);
+
+    for (int ln = coarsest_ln; ln <= finest_ln; ++ln)
+    {
+        Pointer<PatchLevel<NDIM> > level = patch_hierarchy->getPatchLevel(ln);
+        for (PatchLevel<NDIM>::Iterator p(level); p; p++)
+        {
+            Pointer<Patch<NDIM> > patch = level->getPatch(p());
+            const Box<NDIM>& patch_box = patch->getBox();
+            Pointer<CellData<NDIM, double> > D_data = patch->getPatchData(D_idx);
+            for (Box<NDIM>::Iterator it(patch_box); it; it++)
+            {
+                CellIndex<NDIM> ci(it());
+
+                // Get physical coordinates
+                IBTK::Vector X = IBTK::Vector::Zero();
+                Pointer<CartesianPatchGeometry<NDIM> > patch_geom = patch->getPatchGeometry();
+                const double* patch_X_lower = patch_geom->getXLower();
+                const Index<NDIM>& patch_lower_idx = patch_box.lower();
```

Comment:
Please always preface this with `SAMRAI::hier` since both SAMRAI and Eigen define an `Index` template.